api/node/index.ts

```diff
@@ -905,7 +905,7 @@ class EventLoop {
         // can do right now.
         const nodejsPollInterval = 16;
         let id = setInterval(() => {
-            if (napi.processEvents() == napi.ProcessEventsResult.Exited || this.#quit_loop) {
+            if ((napi.processEvents() == napi.ProcessEventsResult.Exited && quitOnLastWindowClosed) || this.#quit_loop) {
```

Comment:
Instead of adding a condition here, I think it would be cleaner to do this once at the beginning in Rust:

```rust
    i_slint_backend_selector::with_platform(|b| {
        #[allow(deprecated)]
        b.set_event_loop_quit_on_last_window_closed(quit_on_last_window_closed);
    })
```